pkg/metrics/publisher/prometheus/prometheus.go

```diff
@@ -0,0 +1,143 @@
+// Package prometheus provides suppoert for sending metrics to prometheus.
+package prometheus
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log/slog"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/dimfeld/httptreemux/v5"
+)
+
+// Exporter implements the prometheus exporter support.
+type Exporter struct {
+	log    *slog.Logger
+	server http.Server
+	data   map[string]any
+	mu     sync.Mutex
+}
+
+// New constructs an Exporter for use.
+func New(log *slog.Logger, host string, route string, readTimeout, writeTimeout time.Duration, idleTimeout time.Duration) *Exporter {
+	mux := httptreemux.NewContextMux()
+
+	exp := Exporter{
+		log: log,
+		server: http.Server{
+			Addr:         host,
+			Handler:      mux,
+			ReadTimeout:  readTimeout,
+			WriteTimeout: writeTimeout,
+			IdleTimeout:  idleTimeout,
+		},
+	}
+
+	mux.Handle(http.MethodGet, route, exp.handler)
+
+	go func() {
+		ctx := context.Background()
+
+		log.InfoContext(ctx, "prometheus", "status", "API listening", "host", host)
+
+		if err := exp.server.ListenAndServe(); err != nil {
+			log.ErrorContext(ctx, "prometheus", "msg", err)
```

Comment:
this error should ideally be propagate to the caller, otherwise the container would start but then it won't be fully functional and the only way to determine it would be by looking at the logs.